### PR TITLE
add support for ghes

### PR DESCRIPTION
**Pull request by marcosnils**
Signed-off-by: Marcos Lilljedahl <marcosnils@gmail.com>

### CODE REVIEW

**Comment by marcosnils** [comment]
cc @sandzwerg would you give it a try?

You need to set the `GHES_*` env variables for this to work. Usually the `UPLOAD_URL` and the `BASE_URL` should be the same one, but some GHES deployment use different ones.

File: pkg/providers/github.go [hunk 1]
```diff
@@ -115,12 +114,34 @@ func newGitHub(u *url.URL) (Provider, error) {
...
+
+	if len(gbu) > 0 && len(guu) > 0 && len(gau) > 0 {
+		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		))
+	} else if token != "" {
 		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
 			&oauth2.Token{AccessToken: token},
 		))
 	}
```
Comment (sirlatrom) [inline pkg/providers/github.go:133]: Why not combine these conditions into one?
```suggestion

	if len(gbu) > 0 && len(guu) > 0 && len(gau) > 0 || token != "" {
		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
			&oauth2.Token{AccessToken: token},
		))
	}
```

**Review by sirlatrom — CHANGES_REQUESTED** [review]

File: pkg/providers/github.go (hunk 1 above)
Comment (marcosnils) [reply pkg/providers/github.go:133]: because I made a mistake and I missed setting the correct token in the `valid` if statement clause :). Fixing

**Comment by marcosnils** [comment]
@sirlatrom mind :eye: one more time? :pray:

**Review by sirlatrom — APPROVED** [review]
LGTM 👍

**Pull request merged**